src/webui/api/torrentscontroller.cpp

```diff
@@ -679,6 +706,58 @@ void TorrentsController::removeTrackersAction()
         torrent->forceReannounce();
 }
 
+void TorrentsController::addPeersAction()
+{
+    checkParams({"hash", "peers"});
+
+    const QString hash = params()["hash"];
+    const QStringList peers = params()["peers"].split('|');
+
+    BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
+    if (!torrent)
+        throw APIError(APIErrorType::NotFound);
```

Comment:
I would also remove the controversial cases of error-handling (we can implement them in separate PR).

Reply:
Agreed, the new response will handle it far better than I can here.